Add KLang

Loidbae opened the pull request:
# KLang

Lets the user change layout and variant of external keyboards connected via bluetooth and/or usb. Currently there is no way to change the default layout of an external keyboard, for example I have a german keyboard which means my layout is "qwertz". Currently there is no other way to change it besides a workaround, which I implemented with this plugin.

Read more on the main page https://github.com/Loidbae/SDH-KLang

## Checklist:

- [ ] I am the original author of this plugin.
- [ ] I made my code efficient and readable.
- [ ] I have verified that my plugin works properly on the latest versions of SteamOS and decky-loader.
- [ ] I have verified that my plugin is unique and does not have the same functionality as another plugin on the store.

TrainDoctor reviewed (commented):
General code-review so that the plugin can go to the store:

Overall looks good an no issues with the underlying code, except for one glaring issue:
Hardcoded file paths. Decky Loader exports quite a few utility functions to determine the current active user to allow plugins to run on other systems with minimal effort and hardcoding the file paths to the deck prevents this plugin from running on other systems. The plugin is approved to go to the testing store, but please keep this in mind before the plugin goes to the main store.

Loidbae commented:
The paths are as now dynamic as they can be, I also added a logo and changed some details in plugin.json and package.json. Yes the logger Path is still hardcoded but I just use it for debugging. :smile:

Loidbae commented:
On second thought it would be better if the logger path was also dynamic to avoid future crashes, so I changed that. :sweat_smile:

TrainDoctor commented:
> JPGs have been swapped out for pngs

Please reduce the length of your description in package.json

Loidbae commented:
Reduced the package description also the plugin description, sorry about that, haha

GeneStarwind09 reviewed (approved):
Tested with a USB keyboard on beta branch. Successfully changed from the default (USA) layout to UK.
